PicPi/FullImageViewController.swift

```diff
@@ -0,0 +1,101 @@
+//  Created by Mustafa T. Mohammed on 1/2/21.
+//
+
+import UIKit
+import Promises
+
+class FullImageViewController: UIViewController {
+  private let deleteButton = customBtnRoundCornerBlueWithShadow(type: .custom)
+  private var margins: UILayoutGuide!
+  private var photo: Photo
+  
+  init(photo: Photo) {
+    self.photo = photo
+    super.init(nibName: nil, bundle: nil)
+  }
+  
+  @available(*, unavailable)
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    margins = view.layoutMarginsGuide
+    view.backgroundColor = .white
+    title = photo.filename
+    
+    setupDeleteButton()
+    setupImageView()
+  }
+  
+  private func setupImageView() {
+    let imageView = UIImageView()
+    imageView.isUserInteractionEnabled = true
+    imageView.translatesAutoresizingMaskIntoConstraints = false
+    imageView.contentMode = .scaleAspectFit
+    imageView.frame = CGRect(x: 0, y: 0, width: view.frame.width, height: view.frame.height)
+    /// download full image from Pi
+    imageView.kf.setImage(with: photo.fullImage)
+    view.addSubview(imageView)
+    
+    NSLayoutConstraint.activate(
+      [
+        imageView.leadingAnchor.constraint(equalTo: margins.leadingAnchor),
+        imageView.trailingAnchor.constraint(equalTo: margins.trailingAnchor),
+        imageView.topAnchor.constraint(equalTo: margins.topAnchor ),
+        imageView.bottomAnchor.constraint( equalTo: deleteButton.topAnchor ),
+      ]
+    )
+  }
+  
+  private func setupDeleteButton() {
+    view.addSubview(deleteButton)
+    ///add action when search button pressed
+    deleteButton.addTarget(self,
+                           action: #selector(deleteButtonPressed),
+                           for: .touchUpInside)
+    deleteButton.setTitle("Delete",
+                          for: .normal)
+    deleteButton.sizeToFit()
+    deleteButton.translatesAutoresizingMaskIntoConstraints = false
+    NSLayoutConstraint.activate(
+      [
+        deleteButton.centerXAnchor.constraint(equalTo: margins.centerXAnchor),
+        deleteButton.centerYAnchor.constraint(
+          equalTo : margins.bottomAnchor,
+          constant:  -deleteButton.frame.height),
+        deleteButton.widthAnchor.constraint(
+          equalToConstant: 100)
+      ]
+    )
+    
+  }
+  
+  @objc func deleteButtonPressed(sender: UIButton!) {
+    
+    let alert = UIAlertController(
+      title: nil,
+      message: "Delete this photo :( ?",
+      preferredStyle: .alert
+    )
+    let yesAction = UIAlertAction(title: "Yes :(", style: .default) { _ in
+      self.photo.removeFromFrame().then { [weak self] (_)   in
+        _ = self?.navigationController?.popViewController(animated: true)
+        
+      }.catch { (error) in
+        print(error)
+      }
+    }
+    let noAction = UIAlertAction(title: "No :)", style: .default )
+    
+    alert.addAction(yesAction)
+    alert.addAction(noAction)
```

Comment:
Do we have an extension for alerts for two buttons?

Reply:
Fixed!